app/src/main/java/io/intrepid/skeleton/base/BaseFragment.java

```diff
@@ -0,0 +1,92 @@
+package io.intrepid.skeleton.base;
+
```

Comment:
`   @CallSuper
    @Override
    public void onPause() {
        if (unBindOnPause()) {
            presenter.unBind();
        }
        super.onPause();
    }

    /**
     * Override this to false to implement a different way of unbinding views with Android's lifecycle methods
     */
    public Boolean unBindOnPause() {
        return true;
    }`

We added a shim to allow users to unbind in onStop or other lifecycle methods. The use case was that when creating a Dialog (not DialogFragment), the lifecycle of the dialog is controlled by the activity and `onPause` is called while instantiating. This unbinds the presenter while the dialog is shown and may inhibit logic that someone has written in their listeners.